src/main/java/org/fogbowcloud/sebal/parsers/WeatherStation.java

```diff
@@ -140,136 +83,119 @@ protected String readFullRecord(Date date, List<JSONObject> stations, int number
-		String relativeHumidity = stationDataRecord
-				.optString(SEBALAppConstants.JSON_RELATIVE_HUMIDITY);
-		String minTemp = stationDataRecord
-				.optString(SEBALAppConstants.JSON_MIN_TEMPERATURE);
-		String maxTemp = stationDataRecord
-				.optString(SEBALAppConstants.JSON_MAX_TEMPERATURE);
-		String solarRad = stationDataRecord
-				.optString(SEBALAppConstants.JSON_SOLAR_RADIATION);
-
-		if (closestRecord.optString(SEBALAppConstants.JSON_STATION_WIND_SPEED)
-				.isEmpty()
-				|| closestRecord
-						.optString(SEBALAppConstants.JSON_STATION_WIND_SPEED) == null) {
-			windSpeed = "NA";
-		}
-		if (closestRecord.optString(SEBALAppConstants.JSON_AIR_TEMPERATURE)
-				.isEmpty()
-				|| closestRecord
-						.optString(SEBALAppConstants.JSON_AIR_TEMPERATURE) == null) {
-			airTemp = "NA";
-		}
-		if (closestRecord
-				.optString(SEBALAppConstants.JSON_DEWPOINT_TEMPERATURE)
-				.isEmpty()
-				|| closestRecord
-						.optString(SEBALAppConstants.JSON_DEWPOINT_TEMPERATURE) == null) {
-			dewTemp = "NA";
-		}
-		if (closestRecord.optString(SEBALAppConstants.JSON_AVG_AIR_TEMPERATURE)
-				.isEmpty()
-				|| closestRecord
-						.optString(SEBALAppConstants.JSON_AVG_AIR_TEMPERATURE) == null) {
-			avgAirTemp = "NA";
-		}
-		if (closestRecord.optString(SEBALAppConstants.JSON_RELATIVE_HUMIDITY)
-				.isEmpty()
-				|| closestRecord
-						.optString(SEBALAppConstants.JSON_RELATIVE_HUMIDITY) == null) {
-			relativeHumidity = "NA";
-		}
-		if (closestRecord.optString(SEBALAppConstants.JSON_MIN_TEMPERATURE)
-				.isEmpty()
-				|| closestRecord
-						.optString(SEBALAppConstants.JSON_MIN_TEMPERATURE) == null) {
-			minTemp = "NA";
-		}
-		if (closestRecord.optString(SEBALAppConstants.JSON_MAX_TEMPERATURE)
-				.isEmpty()
-				|| closestRecord
-						.optString(SEBALAppConstants.JSON_MAX_TEMPERATURE) == null) {
-			maxTemp = "NA";
-		}
-		if (closestRecord.optString(SEBALAppConstants.JSON_SOLAR_RADIATION)
-				.isEmpty()
-				|| closestRecord
-						.optString(SEBALAppConstants.JSON_SOLAR_RADIATION) == null) {
-			solarRad = "NA";
+	protected boolean hasRecord(JSONArray stationData, String key, String value) {
+		boolean result = false;
+		for (int i = 0; i < stationData.length() && !result; i++) {
+			JSONObject stationDataRecord = stationData.optJSONObject(i);
+			
+			if (stationDataRecord.optString(key).equals(value)) {
+				result = true;
+			}
 		}
-
-		toReturn.append(stationId + ";" + dateValue + ";" + timeValue + ";"
-				+ latitude + ";" + longitude + ";" + windSpeed + ";" + airTemp
-				+ ";" + dewTemp + ";" + avgAirTemp + ";" + relativeHumidity
-				+ ";" + minTemp + ";" + maxTemp + ";" + solarRad + ";\n");
+		return result;
 	}
 
-	public double zx(double lat, double lon) {
-		if (properties.getProperty("altitude_sensor_velocidade") != null) {
-			return Double.parseDouble(properties
-					.getProperty("altitude_sensor_velocidade"));
+	protected void windSpeedCorrection(JSONArray stationData) {
```

Comment:
hum ... I have a felling this is not good. Maybe it is just a bias but it prefer return copies of things instead of changing the parameters. As I did not know much about this codebase, ok

Reply:
Done, review!